contrib/node/tests/python/pants_test/contrib/node/tasks/test_apache_thrift_node_gen.py

```diff
@@ -0,0 +1,131 @@
+from textwrap import dedent
+from pants.contrib.node.tasks.apache_thrift_node_gen import ApacheThriftNodeGen
+from pants.contrib.node.targets.node_thrift_library import NodeThriftLibrary
+from pants.contrib.node.targets.node_module import NodeModule
+from pants_test.task_test_base import TaskTestBase
+
+
+class ApacheThriftNodeGenTest(TaskTestBase):
+
+  @classmethod
+  def task_type(cls):
+    return ApacheThriftNodeGen
+
+  def generate_multiple_thrift_targets(self, node_thrift_library):
+    context = self.context(target_roots=[node_thrift_library])
+    apache_thrift_gen = self.create_task(context)
+    apache_thrift_gen.execute()
+
+    def is_synthetic_node_library(target):
```

Comment:
Looks like you test the target here, perhaps you don't need the previous node_thrift_library test

Reply:
This tests the codegen, NodeThriftLibrary test just tests the class, it has similar tests to node_nolule